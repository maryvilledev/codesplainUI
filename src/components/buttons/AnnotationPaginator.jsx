import React, { PropTypes } from 'react';
import { IconButton } from 'material-ui';
import Previous from 'material-ui/svg-icons/navigation/arrow-back';
import Next from 'material-ui/svg-icons/navigation/arrow-forward';

const AnnotationPaginator = ({ hasNextAnnotation, getNextAnnotation,
  hasPrevAnnotation, getPreviousAnnotation, style }) => (
    <div style={style}>
      <IconButton
        disabled={!hasPrevAnnotation}
        id="previous-annotation"
        onTouchTap={getPreviousAnnotation}
        tooltip="Previous Annotation"
        tooltipPosition="top-right"
        touch
      >
        <Previous />
      </IconButton>
      <IconButton
        disabled={!hasNextAnnotation}
        id="next-annotation"
        onTouchTap={getNextAnnotation}
        tooltip="Next Annotation"
        tooltipPosition="top-right"
        touch
      >
        <Next />
      </IconButton>
    </div>
  );

AnnotationPaginator.propTypes = {
  hasNextAnnotation: PropTypes.bool.isRequired,
  hasPrevAnnotation: PropTypes.bool.isRequired,
  getNextAnnotation: PropTypes.func.isRequired,
  getPreviousAnnotation: PropTypes.func.isRequired,
};

export default AnnotationPaginator;
