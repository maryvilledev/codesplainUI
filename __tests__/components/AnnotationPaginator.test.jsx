import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import AnnotationPaginator from '../../src/components/buttons/AnnotationPaginator';

const defaultProps = {
  getNextAnnotation: jest.fn(),
  getPreviousAnnotation: jest.fn(),
  hasNextAnnotation: false,
  hasPrevAnnotation: false,
};

describe('<AnnotationPanel />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshots', () => {
    it('matches snapshot', () => {
      const wrapper = shallowWithContext(<AnnotationPaginator {...defaultProps} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
