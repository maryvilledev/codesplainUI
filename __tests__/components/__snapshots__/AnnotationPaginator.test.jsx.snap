exports[`<AnnotationPanel /> snapshots matches snapshot 1`] = `
<div>
  <IconButton
    disableTouchRipple={false}
    disabled={true}
    iconStyle={Object {}}
    id="previous-annotation"
    onTouchTap={[Function]}
    tooltip="Previous Annotation"
    tooltipPosition="top-right"
    touch={true}>
    <NavigationArrowBack />
  </IconButton>
  <IconButton
    disableTouchRipple={false}
    disabled={true}
    iconStyle={Object {}}
    id="next-annotation"
    onTouchTap={[Function]}
    tooltip="Next Annotation"
    tooltipPosition="top-right"
    touch={true}>
    <NavigationArrowForward />
  </IconButton>
</div>
`;
